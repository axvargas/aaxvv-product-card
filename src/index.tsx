export * from './components';

// interfaces
export {Product} from './interfaces/interfaces';