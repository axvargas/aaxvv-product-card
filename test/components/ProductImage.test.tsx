import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard, ProductImage } from '../../src/components'
import { product2 } from '../data/products';


describe('ProductImage', () => {
  test('Must render with the personalized image', () => {
    const wrapper = renderer.create(<ProductImage img={product2.img} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
  test('Must render the ProductCard with the image of the product', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          () => (
            <ProductImage />
          )
        }
      </ ProductCard >
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
