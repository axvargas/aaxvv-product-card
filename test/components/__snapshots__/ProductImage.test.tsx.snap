// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Must render the ProductCard with the image of the product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffe-mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Must render with the personalized image 1`] = `
<img
  alt="coffe-mug"
  className="productImg undefined"
  src="./coffee-mug.png"
/>
`;
